optimum/exporters/onnx/utils.py

```diff
@@ -131,15 +128,12 @@ def get_decoder_models_for_export(
         `Dict[str, Tuple[Union[`PreTrainedModel`, `TFPreTrainedModel`], `OnnxConfig`]: A Dict containing the model and
         onnx configs for the encoder and decoder parts of the model.
     """
-    models_for_export = dict()
+    models_for_export = {}
 
-    models_for_export["decoder_model"] = (
-        model,
-        config.__class__(model.config, task=config.task, use_past=False, use_present_in_outputs=True),
-    )
+    models_for_export["decoder_model"] = (model, config.with_behavior("decoder", use_past=False))
```

Comment:
With behaviour is for seq2seq models, this will not work for models such as gpt2.